### ISSUE TITLE
Socket doesn't seem to close after unexpected disconnect/reconnect

### ISSUE THREAD

**Issue by bachya** [miguelgrinberg/python-socketio#285]
This report comes via [`aioambient`](https://github.com/bachya/aioambient), which uses this library to integrate a Socket.IO-based Ambient Weather socket for Home Assistant: https://github.com/home-assistant/home-assistant/issues/22636

The user reports that after the socket unexpectedly disconnects and reconnects, the old socket doesn't get cleaned up – below is a `netstat` output for connections to the Ambient Weather Socket.IO instance:

```
$ netstat -p | grep '104.31' | more
tcp        0      0 home:48313              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48278              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48344              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44242              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44277              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48305              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48297              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48291              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48315              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48298              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44293              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48310              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44297              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44280              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48323              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48259              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48336              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44281              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48286              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44316              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44288              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44273              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:48321              104.31.82.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44283              104.31.83.184:https     ESTABLISHED 7650/python3
tcp        0      0 home:44321              104.31.83.184:https     ESTABLISHED 7650/python3
```

FWIW, `aioambient` uses version 3.1.2 of `python-socketio`.

**EDIT:** possible duplicate of https://github.com/miguelgrinberg/python-socketio/issues/267? If so, looks like an upgrade to 4.0.0 will help?

**Comment by miguelgrinberg** [comment]
The fix for the unclosed sessions with aiohttp is in python-engineio. Do you know what version of this package presents this problem? The fix for this problem was introduced in v3.4.4, so I would suggest an upgrade if it is currently older.

**Comment by bachya** [comment]
Got it – I will upgrade and report back if needed. Thanks!
